scripts/generate_query_config.pl

```diff
@@ -100,13 +102,42 @@
     close(CONFIG_FILE);
 }
 
+# We need to include all the headers with public APIs in case they
+# define a macro to its default value when that configuration is not
+# set in a header included by build_info.h (crypto_config.h,
+# mbedtls_config.h, *adjust*.h). Some module-specific macros are set
+# in that module's header. For simplicity, include all headers, with
+# some ad hoc knowledge of headers that are included by other headers
+# and should not be included directly. We don't include internal headers
+# because those should not define configurable macros.
+my @header_files = ();
+my @header_roots = qw(
+                         include
+                         tf-psa-crypto/include
+                         tf-psa-crypto/drivers/builtin/include
```

Comment:
Note: I was a bit surprised by this one. But I guess it's possible for say `aes.c` to contain a default definition for some `MBEDTLS_AES_xxx` macro that we'd want to query with this program, so OK.

Reply:
In https://github.com/Mbed-TLS/mbedtls/pull/10325/files#diff-2a5263f7c0d64de75449e9461c260e10f4a7ac847123c7b93b8ebad174dcf737R22 I tried to minimize the list while keeping `query_compile_time_config -l` unchanged. I had to keep `ctr_drbg.h`,  `ecp.h`, `entropy.h`, `hmac_drbg.h`, `rsa.h`.

The fact that I had to keep some private headers was part of why I prefer the approach here at least at this time: this way we can reorganize private headers in crypto without having to worry about this piece of mbedtls.